README.md

```diff
@@ -88,43 +134,101 @@ async function main(): Promise<void> {
-    },
-    onError: (context: EventContext) => {
-      const receiverError = context.receiver && context.receiver.error;
-      if (receiverError) {
-        console.log("[%s] An error occurred for receiver '%s': %O.",
-          connection.id, receiverName, receiverError);
-      }
-    },
     onSessionError: (context: EventContext) => {
       const sessionError = context.session && context.session.error;
       if (sessionError) {
-        console.log("[%s] An error occurred for session of receiver '%s': %O.",
+        console.log(">>>>> [%s] An error occurred for session of receiver '%s': %O.",
           connection.id, receiverName, sessionError);
       }
     }
   };
 
   await connection.open();
   const receiver: Receiver = await connection.createReceiver(receiverOptions);
+  receiver.on(ReceiverEvents.message, (context: EventContext) => {
+    console.log("Received message: %O", context.message);
+  });
+  receiver.on(ReceiverEvents.receiverError, (context: EventContext) => {
+    const receiverError = context.receiver && context.receiver.error;
+    if (receiverError) {
+      console.log(">>>>> [%s] An error occurred for receiver '%s': %O.",
+        connection.id, receiverName, receiverError);
+    }
+  });
   // sleeping for 2 mins to let the receiver receive messages and then closing it.
   await delay(120000);
   await receiver.close();
   await connection.close();
 }
 
 main().catch((err) => console.log(err));
+```
 
+## Design 
+The library provides an easy to use promisfied API for creating connection, session, sender and 
+receiver links. One can provide event handlers as optional properties while creating entities or 
+can use the standard event emitter pattern to add listeners to the created entities.
+
+For implementing the Event Emitter pattern, we considered different approaches like:
+**extending rhea's objects** (Connection, Session, Sender, Receiver). However, this has an overhead 
+of ensuring that there are no naming conflicts in the method names. Selecting different names for 
+operations like `open` and `close` would add more confusion.
+
+Hence we settled with, **wrapping rhea's objects** under our own equivalent objects. To ensure that 
+user's can efficiently use different event emitter methods like `.once()`, `.on()`,
+`.prependListeners()`, etc. we decided to add listeners to rhea's objects 
+(connection, session, sender, receiver) for all the possible events that those objects can 
+emit. From those listeners, equivalent `rhea-promise` objects emit the same events with same 
+arguments as one would expect from an event emitted by one of the above mentioned objects from 
+`rhea`. User's can add/remove listeners to `rhea-promise` objects. Whenever an event is emitted 
+from an object from `rhea`, it's equivalent counterpart in `rhea-romise` would emit the same event.
+
+`rhea` deals little differently with *_error and *_close events. If an event listener is **not added** 
```

Comment:
the \* in \*_error and \*_close should be escaped with a backslash to avoid it being interpreted as markdown formatting. there are one or two places below with the same setup